src/ageml/modelling.py

```diff
@@ -367,105 +372,132 @@ def fit_age(self, X, y):
+                original_pipeline = copy.deepcopy(self.pipeline)
+                point_pipeline = original_pipeline.set_params(**grid_point)
+                pipelines.append(point_pipeline)
+            # Variables of interest
+            pred_age = np.zeros(y.shape[0])
+            corrected_age = np.zeros(y.shape[0])
+            best_split_mae = 1e10
+            mae_means_test = []
+            # Loop through the pipelines, and then loop through the CV splits
+            for cv_pipeline in pipelines:
+                print(f"\nRunning CV splits with pipeline:\n{cv_pipeline}")
+                temp_pred_age = np.zeros(y.shape[0])
+                temp_corr_age = np.zeros(y.shape[0])
+                split_metrics_train = []
+                split_metrics_test = []
+                kf_hyperopt = model_selection.KFold(n_splits=self.CV_split, random_state=self.seed, shuffle=True)
+                for i, (train, test) in enumerate(kf_hyperopt.split(X)):
+                    X_train, X_test = X[train], X[test]
+                    y_train, y_test = y[train], y[test]
+
+                    # Train model with the hyperparameters
+                    cv_pipeline.fit(X_train, y_train)
+
+                    # Predictions on train and test set
+                    y_pred_train = cv_pipeline.predict(X_train)
+                    y_pred_test = cv_pipeline.predict(X_test)
+
+                    # Metrics in Train and Test sets
+                    split_metrics_train.append(self.calculate_metrics(y_train, y_pred_train))
+                    split_metrics_test.append(self.calculate_metrics(y_test, y_pred_test))
+
+                    # Print metrics
+                    print("\nFold N: %d" % (i + 1))
+                    print("Train: MAE %.2f, RMSE %.2f, R2 %.3f, p %.3f" % split_metrics_train[i])
+                    print("Test: MAE %.2f, RMSE %.2f, R2 %.3f, p %.3f" % split_metrics_test[i])
+
+                    # Fit and apply age-bias correction
+                    self.fit_age_bias(y_train, y_pred_train)
+                    y_pred_test_no_bias = self.predict_age_bias(y_test, y_pred_test)
+
+                    # Save results of hold out
+                    temp_pred_age[test] = y_pred_test
+                    temp_corr_age[test] = y_pred_test_no_bias
+
+                # Compute the mean of scores over all CV splits
+                mean_score_test = np.mean(split_metrics_test, axis=0)[0]  # Mean of MAE
+                mae_means_test.append(mean_score_test)
+
+                # If the mean MAE is better than the previous best, save the results
+                if mean_score_test < best_split_mae:
+                    best_split_mae = mean_score_test
+                    pred_age = copy.deepcopy(temp_pred_age)
+                    corrected_age = copy.deepcopy(temp_corr_age)
+
+                # CV Summary of the selected hyperparameters
+                print("\nSummary metrics over all CV splits:")
```

Comment:
I believe here you are printing the metrics from the last CV point and not the best CV point. This should be changed.